### ISSUE TITLE
balenaEtcher (upstream needs to update to electron 13+)

### ISSUE THREAD

**Issue by Crilum** [Botspot/pi-apps#1828]
### Confirmations

- [X] I searched for similar bug reports in https://github.com/Botspot/pi-apps/issues and found none were relevant.
- [X] I tried updating Pi Apps and the issue is still not fixed. [How to update Pi Apps?](https://github.com/Botspot/pi-apps#to-update-pi-apps)

### What happened?

Etcher fails with seccomp sandbox error (see https://github.com/Botspot/pi-apps/issues/1828#issuecomment-1128282925 for upstream issues)

### Description

Etcher has been broken for some time now..
See logs below.

This might have something to do with #1

### What is your operating system?

Raspberry Pi OS 11 (Bullseye)

### (Optional) Error log? Terminal output? Debug messages?

```bash
$ /opt/balenaEtcher/balena-etcher-electron
[Axios v1.7.9+5945ab1f] Transitional option 'clarifyTimeoutError' has been deprecated since v1.0.0 and will be removed in the near future
[Axios v1.7.9+5945ab1f] Transitional option 'forcedJSONParsing' has been deprecated since v1.0.0 and will be removed in the near future
[Axios v1.7.9+5945ab1f] Transitional option 'silentJSONParsing' has been deprecated since v1.0.0 and will be removed in the near future
../../sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.cc:**CRASHING**:seccomp-bpf failure in syscall 0403
../../sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.cc:**CRASHING**:seccomp-bpf failure in syscall 0403
../../sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.cc:**CRASHING**:seccomp-bpf failure in syscall 0403
../../sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.cc:**CRASHING**:seccomp-bpf failure in syscall 0403
../../sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.cc:**CRASHING**:seccomp-bpf failure in syscall 0403
../../sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.cc:**CRASHING**:seccomp-bpf failure in syscall 0403
../../sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.cc:**CRASHING**:seccomp-bpf failure in syscall 0403
../../sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.cc:**CRASHING**:seccomp-bpf failure in syscall 0403
../../sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.cc:**CRASHING**:seccomp-bpf failure in syscall 0403
[3191:0516/164142.592492:FATAL:gpu_data_manager_impl_private.cc(415)] GPU process isn't usable. Goodbye.
/opt/balenaEtcher/balena-etcher-electron: line 10:  3191 Trace/breakpoint trap   "${script_dir}"/balena-etcher-electron.bin "$@"
```

**Comment by Crilum** [comment]
Adding `--disable-seccomp-filter-sandbox` fixes the problem.

**Comment by ryanfortner** [comment]
Interesting. I built the v1.7.9 debs using Debian Buster. I’ll try with Bullseye and see if there is any difference.

**Comment by Crilum** [comment]
Yeah, I can do that.

**Comment by Crilum** [comment]
On bullseye, etcher just stalls after selecting an image..

**Comment by ryanfortner** [comment]
> On bullseye, etcher just stalls after selecting an image..

did you select an .xz image? I remember having problems with etcher using those in the past.

**Comment by Crilum** [comment]
Yeah, I did actually.. It seemed to work before I did that.

**Comment by Crilum** [comment]
It works fine in buster. I noticed during the install that chmod failed to, well, chmod `/opt/balenaEtcher/chrome-sandbox`. It said there was no such file or directory.

```
$ /opt/balenaEtcher/balena-etcher-electron
[Axios v1.7.9+5945ab1f] Transitional option 'clarifyTimeoutError' has been deprecated since v1.0.0 and will be removed in the near future
[Axios v1.7.9+5945ab1f] Transitional option 'forcedJSONParsing' has been deprecated since v1.0.0 and will be removed in the near future
[Axios v1.7.9+5945ab1f] Transitional option 'silentJSONParsing' has been deprecated since v1.0.0 and will be removed in the near future
ready-to-show: 4.397s
```

**Comment by ryanfortner** [comment]
> I noticed during the install that chmod failed to, well, chmod `/opt/balenaEtcher/chrome-sandbox`. It said there was no such file or directory.

That is safe to ignore.

I will add this to my to-do list and get new debs released as soon as I can.

**Comment by Crilum** [comment]
I just built on bullseye, we still have the seccomp-sandbox error:
```
webpack 5.11.0 compiled successfully in 41623 ms
do you want to run a test of etcher to see if compilation worked [y/n]? y
close etcher window to continue to packaging.
waiting for 5 seconds.....
Starting Etcher...

> balena-etcher@1.7.9 start
> ./node_modules/.bin/electron .

[Axios v1.7.9] Transitional option 'clarifyTimeoutError' has been deprecated since v1.0.0 and will be removed in the near future
[Axios v1.7.9] Transitional option 'forcedJSONParsing' has been deprecated since v1.0.0 and will be removed in the near future
[Axios v1.7.9] Transitional option 'silentJSONParsing' has been deprecated since v1.0.0 and will be removed in the near future
../../sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.cc:**CRASHING**:seccomp-bpf failure in syscall 0403
../../sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.cc:**CRASHING**:seccomp-bpf failure in syscall 0403
../../sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.cc:**CRASHING**:seccomp-bpf failure in syscall 0403
../../sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.cc:**CRASHING**:seccomp-bpf failure in syscall 0403
../../sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.cc:**CRASHING**:seccomp-bpf failure in syscall 0403
../../sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.cc:**CRASHING**:seccomp-bpf failure in syscall 0403
../../sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.cc:**CRASHING**:seccomp-bpf failure in syscall 0403
../../sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.cc:**CRASHING**:seccomp-bpf failure in syscall 0403
../../sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.cc:**CRASHING**:seccomp-bpf failure in syscall 0403
[8023:0516/181257.120324:FATAL:gpu_data_manager_impl_private.cc(415)] GPU process isn't usable. Goodbye.
/home/pi/src/etcher/etcher/node_modules/electron/dist/electron exited with signal SIGTRAP
Etcher failed to start!
```

Here's the deb:

[balena-etcher-electron_1.7.9%2B5945ab1f_armv7l.deb](https://github.com/Crilum/stuff/raw/main/balena-etcher-electron_1.7.9%2B5945ab1f_armv7l.deb)

**Comment by theofficialgman** [comment]
I have already told you before, this is an upstream etcher issue. the version of electron use with etcher (12.2.3) is very very old and has this bug. they need to upgrade their codebase to one of the versions that doesn't have this issue.

**Comment by Botspot** [comment]
Would temporarily downgrading balenaEtcher solve this problem?

**Comment by Botspot** [comment]
Would it be worthwhile to add the `--disable-seccomp-filter-sandbox` flag for the time being?

**Comment by Crilum** [comment]
@Botspot I think it would. Otherwise Etcher is completely broken for all Bullseye users..

**Comment by Botspot** [comment]
> @Botspot I think it would. Otherwise Etcher is completely broken for all Bullseye users..

I've opened an issue on @Itai-Nelken's repo.
https://github.com/Itai-Nelken/Etcher-arm-32-64/issues/6

**Comment by ryanfortner** [comment]
@Crilum this issue should be fixed with https://github.com/Botspot/pi-apps/pull/1838. The desktop entry in the new debs has `--disable-seccomp-filter-sandbox` included in the run command.

**Comment by theofficialgman** [comment]
lets keep this open for tracking upstream electron, since this is only a temporary fix until they update

**Comment by Botspot** [comment]
Has balena updated their Electron version yet?

**Comment by theofficialgman** [comment]
Etcher hasn't had a commit since April.
I think balena may have abandoned the software

**Comment by theofficialgman** [comment]
upstream is opposed to fixing their repo <https://github.com/balena-io/etcher/issues/3647#issuecomment-1312772852>

this issue is closed in pi-apps as since the v1.10.0 release I have packaged it with electron ^13.5.0
https://github.com/Botspot/pi-apps/commit/a09e87f20e48188631090496027c42b081489476
https://github.com/Pi-Apps-Coders/files/blob/main/CompileCommands.md#balenaetcher
